storage/innobase/xtrabackup/src/xbcrypt.c

```diff
@@ -504,11 +518,18 @@ mode_encrypt(File filein, File fileout)
 	ivbuf = my_malloc(encrypt_iv_len, MYF(MY_FAE));
 
 	/* now read in data in chunk size, encrypt and write out */
-	chunkbuflen = opt_encrypt_chunk_size;
-	chunkbuf = my_malloc(chunkbuflen, MYF(MY_FAE));
-	while ((bytesread = my_read(filein, chunkbuf, chunkbuflen,
+	chunkbuflen = opt_encrypt_chunk_size + XB_CRYPT_HASH_LEN;
+	chunkbuf = (uchar *) my_malloc(chunkbuflen, MYF(MY_FAE));
```

Comment:
`my_malloc` is a handy wrapper declared in MySQL source code. It takes flags. The `MY_FAE` flag signals treat allocation errors as fatal, i.e. crash if memory could not be allocated. It is the best we can do actually.